src/common/options/CMakeLists.txt

```diff
@@ -104,8 +103,9 @@ add_options(rgw)
 
 add_library(common-options-objs OBJECT
   ${common_options_srcs})
-add_custom_target(legacy-option-headers
-  DEPENDS ${legacy_options_headers})
+
+# Make common-options-objs as a pre-build requirement.
+add_dependencies(prebuild-targets common-options-objs)
```

Comment:
@thomasgoirand hi Thomas, what these targets depend on are the option headers. and we do have a target for building them, why you prefer adding a new target and let it depend on `common-option-objs` instead? can we just s/prebiuld-targets/legacy-option-headers/ ? or you've tried it?